sim/paladin/retribution_talents.go

```diff
@@ -0,0 +1,237 @@
+	})
+}
+
+func (paladin *Paladin) applySanctityOfBattle() {
+	if !paladin.Talents.SanctityOfBattle {
+		return
+	}
+
+	spenderCooldownMod := paladin.AddDynamicMod(core.SpellModConfig{
+		Kind:      core.SpellMod_Cooldown_Flat,
+		ClassMask: SpellMaskCrusaderStrike | SpellMaskDivineStorm,
+	})
+
+	paladin.AddOnCastSpeedChanged(func(_ float64, castSpeed float64) {
+		spenderCooldownMod.UpdateTimeValue(-(4500*time.Millisecond - time.Duration(4500*castSpeed)))
+	})
+
+	core.MakePermanent(paladin.GetOrRegisterAura(core.Aura{
+		Label:      "Sanctity of Battle",
+		ActionID:   core.ActionID{SpellID: 25956},
+		BuildPhase: core.CharacterBuildPhaseAll,
+
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			spenderCooldownMod.Activate()
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			spenderCooldownMod.Deactivate()
+		},
+	}))
+}
+
+func (paladin *Paladin) applySealsOfCommand() {
+	if !paladin.Talents.SealsOfCommand {
+		return
+	}
+
+	// Seals of Command
+	sealsOfCommandProc := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 20424},
+		SpellSchool: core.SpellSchoolHoly,
+		ProcMask:    core.ProcMaskEmpty,
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagIncludeTargetBonusDamage | core.SpellFlagNoOnCastComplete,
+
+		DamageMultiplier: 0.07,
+		CritMultiplier:   paladin.DefaultMeleeCritMultiplier(),
+		ThreatMultiplier: 1.0,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := spell.Unit.MHNormalizedWeaponDamage(sim, spell.MeleeAttackPower())
+			spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMeleeWeaponSpecialHitAndCrit)
+		},
+	})
+
+	core.MakeProcTriggerAura(&paladin.Unit, core.ProcTrigger{
+		Name:           "Seals of Command",
+		ActionID:       core.ActionID{SpellID: 85126},
+		Callback:       core.CallbackOnSpellHitDealt,
+		Outcome:        core.OutcomeLanded,
+		ClassSpellMask: SpellMaskSealOfTruth | SpellMaskSealOfRighteousness | SpellMaskSealOfJustice,
+		ProcChance:     1.0,
+
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			sealsOfCommandProc.Cast(sim, result.Target)
+		},
+	})
+}
+
+func (paladin *Paladin) applySanctifiedWrath() {
+	if paladin.Talents.SanctifiedWrath == 0 {
+		return
+	}
+
+	paladin.AddStaticMod(core.SpellModConfig{
+		ClassMask:  SpellMaskHammerOfWrath,
+		Kind:       core.SpellMod_BonusCrit_Rating,
+		FloatValue: 0.02 * float64(paladin.Talents.SanctifiedWrath) * core.CritRatingPerCritChance,
+	})
+	paladin.AddStaticMod(core.SpellModConfig{
+		ClassMask: SpellMaskAvengingWrath,
+		Kind:      core.SpellMod_Cooldown_Flat,
+		TimeValue: time.Second * 20 * time.Duration(paladin.Talents.SanctifiedWrath),
+	})
+
+	// Hammer of Wrath execute restriction removal is handled in hammer_of_wrath.go
+}
+
+func (paladin *Paladin) applyCommunion() {
+	if !paladin.Talents.Communion {
+		return
+	}
+
+	paladin.PseudoStats.DamageDealtMultiplier *= 1.02
+}
```

Comment:
Is this a personal extra bonus damage on top of the effect that the buff has for the raid?